### PR TITLE
refactor: Delete Visitor_AST.ml

### PR DESCRIPTION
**Pull request by nmote**
The remaining utilities seemed reasonable to put into AST_generic_helpers

Ran the codemod using these steps:
```
semgrep --lang ocaml -e 'Visitor_AST.$X' . --replacement 'AST_generic_helpers.$X' --autofix
find . -name '*.ml*' -exec sed -i -e '/^module.*= Visitor_AST$/d' {} \;
```

Then did a little bit of manual fixup for the cases that got missed.

Test plan: Automated tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by IagoAbal — APPROVED** [review]
I have only reviewed the files that I "own".

**Comment by nmote** [comment]
Thanks, I'll wait for another review before mergin.

**Review by aryx — APPROVED** [review]
Haha! we need to allow autofix replacement string to be empty!
We need to use semgrep for removing the module V = Visitor_AST.
Shame on us for still using sed!

**Pull request merged**
